docs/modules/data-structures/pages/vector-collections.adoc

```diff
@@ -0,0 +1,495 @@
+= Vector Collection
+:page-enterprise: true
+:page-beta: true
+
+== Intro
+
+The primary object for interacting with the vector storage is a Vector Collection. This collection holds information about the vectors and associated metadata (user value).
+
+A collection consists of one or more indexes, all sharing a common metadata storage. Each index represents a distinct vector space. Vectors from different indexes are stored independently, and indexes can have different configurations.
+
+Conceptually, a vector collection resembles a key-value store. Here, the key is a user-defined unique identifier, and the value is an object containing metadata (user value) alongside several vectors — one for each index.
+
+Users are permitted to store any necessary value for work as metadata (user value). This can be additional characteristics of the source data or even the source data itself.
+
+== Configuration
+:stem: latexmath
+Collection configuration can be set dynamically during vector collection creation or statically during cluster configuration. Unlike other data structures, the configuration must be set up before the collection can be used. Below is a table listing all available options:
```

Comment:
for other data structures you can use both `default` and `*`, for vector with the implementation according to the TDD `*` will also work (if defined) as default for all vector collections. However, `*` is not pre-defined, which is reasonable and I do not see any problems with implementing wildcard matching.

Reply:
> currently the code does not perform wildcard-matching. So if a user configures a VectorCollectionConfig with name "vc-*" then vector collection "vc-1" or "vc-2" won't match.

But it works. If the config "vc-*" exists, then we can create collections with names vc-1 and vc-2 that will reuse the vc-* config

Reply:
info about default config and wildcards:
cadf256ce433c4af7bc8c3ae5286eb8a3b8c328e

Reply:
🤦  doh, I was looking at `Config#getVectorCollectionConfigOrNull` instead of its overridden version in `DynamicConfigurationAwareConfig`. So apparently wildcards already work.